### DISCUSSION TITLE
[@types/react] using react v18 with Typescript < v5.1, while allowing string as valid JSX Element

### DISCUSSION THREAD

**Discussion by Liooo** [DefinitelyTyped/DefinitelyTyped#69919] [Issues with a @types package] [answered]
(**Minimum reproducible repo is https://github.com/Liooo/vite-project, just run `yarn install && yarn tsc` to see the error.**)

I want to use react v18 with Typescript < v5.1, while allowing string (and other ReactNode types) to be valid JSX Elements.

Currently it errors like the following (as is probably obvious for the fellows here):

```tsx
const MyComponent = () => 'x'

function App() {
  return (
      <div>
        <MyComponent />
    　　 // ↑
        // error TS2786: 'MyComponent' cannot be used as a JSX component.
        //     Its return type 'string' is not a valid JSX element.
      </div>
  )
}
```

package.json is:

```json
{
  "dependencies": {
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "typescript": "4.9.5" // <= the error disappears with for example "5.1.6"
  }
}
```

Tried tweaking JSX.Element augmentation [like these](https://github.com/Liooo/vite-project/blob/main/src/globals.d.ts) but no luck so far.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
> I want to use react v18 with Typescript < v5.1, while allowing string (and other ReactNode types) to be valid JSX Elements.

This is not possible in TypeScript < 5.1. In order to support this, we had to contribute a feature that's only available starting with TypeScript 5.1: https://github.com/microsoft/TypeScript/pull/51328

**Reply by Liooo** [reply]
hmm got it, thank you 🫡
